.github/workflows/pull_request.yml

```diff
@@ -139,3 +142,30 @@ jobs:
           name: coverage-report
           path: |
             ${{ github.workspace }}/.out/reports/jacoco/codeCoverageReport/
+
+  coverage-comment:
+    runs-on: ubuntu-latest
+    needs: coverage
+    permissions:
+      checks: write
+      contents: read
+      pull-requests: write
```

Comment:
I don't think that this will work: `pull_request` workflows have their permissions always limited, since they are running code from the PR itself. (There is `pull_request_target`, but the protection is intentional.)

If the `pull_request` triggering the `workflow_run` isn't working properly, then we could write the PR number into an artifact and read from there in the `workflow_run` workflow, passing the number to the comment action. See https://github.com/Madrapps/jacoco-report/blob/main/src/action.ts#L92-L101 for how it tries to figure it out -- which is evidently not working -- and the available override.